sfge/src/main/java/com/salesforce/graph/ops/SoqlParserUtil.java

```diff
@@ -215,7 +226,8 @@ private static SoqlQueryInfo getSoqlQueryInfo(
                 isLimit1(query),
                 // Use securityEnforced value of outer query if inner query doesn't have the clause
                 outerIsSecurityEnforced || isSecurityEnforced(query),
-                isOutermost);
+                isOutermost,
+                outerIsUserMode || isUserMode(query));
```

Comment:
I had to add the `|| isUserMode()` call because `outerIsUserMode` is always false.
This was adapted from the treatment of `isSecurityEnforced`.
Question: It looks like the outer query is being treated as an "inner query" of itself; why is it like that?

Reply:
Can we keep `isOutermost` as the last item on the parameter list?

Been a while since I worked on this portion. I could be off, but I remember creating this as a recursive query parser where information is bubbled from inner layer(s) to outer layer. `isOutermost` is set to true when it reaches the outermost layer of the query. Other than that, each layer is handled the same way.